### PR TITLE
[Term Entry] Lua Strings upper()

### PR DESCRIPTION
**Pull request by rhysfirkins**
### Description

 - Added the string.upper() term for Lua
 - Issue #3106 

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/lua/concepts/strings/terms/upper/upper.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+Title: '.upper()' # Required; the file name should be the same as the title, but lowercase, with dashes instead of spaces, and all punctuation removed
+Description: 'Converts all characters in a string to upper case.' # Required; ideally under 150 characters and starts with a present-tense verb (used in search engine results and content previews)
+Subjects: # Please only use Subjects in the subjects.md file (https://github.com/Codecademy/docs/blob/main/documentation/subjects.md). If that list feels insufficient, feel free to create a new Subject and add it to subjects.md in your PR!
+  - 'Code Foundations'
+Tags: # Please only use Tags in the tags.md file (https://github.com/Codecademy/docs/blob/main/documentation/tags.md). If that list feels insufficient, feel free to create a new Tag and add it to tags.md in your PR!
+  - 'Characters'
+  - 'Functions'
+  - 'Strings'
+CatalogContent: # Please use course/path landing page slugs, rather than linking to individual content items. If listing multiple items, please put the most relevant one first
+  - 'learn-lua'
```
Comment (SSwiniarski) [inline content/lua/concepts/strings/terms/upper/upper.md:11]:
```suggestion
  - 'learn-lua'
  - 'paths/computer-science'
```

File: content/lua/concepts/strings/terms/upper/upper.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+> **Note:** The `.upper()` function creates a copy of the specified string to modify, leaving the original unchanged.
+
+```lua
+myString = "Bang!"
+
+print(myString)
+print(string.upper(myString))
+print(myString)
+```
+This code would display the following output:
```
Comment (SSwiniarski) [inline content/lua/concepts/strings/terms/upper/upper.md:35]:
```suggestion

This code would display the following output:
```

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@rhysfirkins I did a first review and left a few comments.

File: content/lua/concepts/strings/terms/upper/upper.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+myString = "Bang!"
+
+print(myString)
+print(string.upper(myString))
+print(myString)
+```
+
+This code would display the following output:
+
+```lua
```
Comment (SSwiniarski) [inline content/lua/concepts/strings/terms/upper/upper.md:40]:
````suggestion
```shell
````

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@rhysfirkins One more comment.

**Review by SSwiniarski — APPROVED** [review]
@rhysfirkins Looks good for a second review.

**Review by caupolicandiaz — APPROVED** [review]
@rhysfirkins @SSwiniarski Made some of the missing changes and a couple of additional edits. Looks good to merge

**Pull request merged**